### ISSUE TITLE
hw06 ready for grading

### ISSUE THREAD

**Issue by bcahn7** [bcahn7/STAT547M-hw-An-Byeongchan#1]
hw06 ready for grading

**Comment by peterwhitman** [comment]
Hi,

Nice job on homework 6! You addressed all three tasks more than adequately. You frequently explored the functionality of packages and functions beyond what was expected, and clearly articulated your process. I especially like that the function you created allows the user to pick which regression type to run. Your homework is also very easy to follow. I like that you have included portions of the prompt and your own comments. This helps the viewer (me) understand your workflow. If the quality of your homework is any indication of your grasp on the material, it seems like you have a good handle on the subject matter.

Good work!

Peter

**Comment by yidie** [comment]
Hi @bcahn7,

- Work with a list
Good job working through the Trump Android Tweets. You have comments and explanation throughout the code, which makes it easy to follow and understand. Application to finance is also well done. It shows your understanding of `map()` functions.

- Writing functions
It would be better if you can have legend labelled on the graph although I can refer back to your code to figure out which one is what method. The function is great! It takes mode and year as inputs and output the parameters after fitting regression. You incorporated three different fitting methods via if...else... statement. One small tip: you can also use `I(lifeExp^2)` or `poly(lifeExp, 2)` as arguments of `lm()` to do the quadratic fit instead of mutating the data.

- Work with Singer data
Good attempts with this task. I am not sure what those warnings mean. But It is nice that you tried and explained what you did and your thoughts.

Overall, good work. You did three tasks, which is more than required. And your progress report is detailed and shows how you tackled the problems. And the md file is well organized according to the instructions. 

Yidie

**Comment by derekcho** [comment]
Hi @bcahn7 ! Here are some comments about your hw06:

-	For your first task, a bit of the work is hard to follow. I think that you should insert some more explanations in your work rather than relying mostly on inline comments. Even though you can get the work done, it takes the reader some time to figure out what you are trying to do each step of the way. For example, when you show a chunk of code and output, the output does not mean much without an interpretation for the reader. Since you don’t provide any further interpretations for that task, it was easy to get lost in what you were trying to do even though it appears to be correct! 
-	Your second task (writing a function) was easier to read and understand compared to your first one, although I think that you could still put in some explanations. I like that you plotted all three linear models using the gapminder data! Your function is written well and does something interesting with the gapminder data. However, it would have been nice to see you include some of the other aspects of functions we discussed in class (default values, error checking, etc.).  In addition, I am curious as to why Reg_life_gdp(“rlm”,2007) gives a different output compared to rlm(gdpPercap ~ lifeExp, gap_2007)
-	For your third task (singer data), you need to include explanations of what you are trying to do and interpret the outputs. Inline comments are not enough! You used revgeocode, but I cannot see the locations so I cannot tell what your results are (until the very end).
-	You should remove warnings from your report, they make it look messy and unorganized. Use the statement warning=FALSE in your header for the chunk of R code in Rmarkdown
-	All in all, you were able to complete each task successfully, but it took quite a bit of effort to understand each step of your work. It is great that you attempted to do 3 tasks instead of the required 2, but I would suggest that you spend a bit of extra time improving the explanations for your tasks. When you write your assignment, you should imagine that you are writing a report for someone who is unfamiliar with the tasks. You produced some good results, but it is important for you to be able to present them too!

Note: your mark (check minus, check, check plus) will be distributed later.
